openff/interchange/_tests/unit_tests/interop/openmm/test_constraints.py

```
@@ -0,0 +1,249 @@
+        },
+    )
+
+    return handler
+
+
+@pytest.fixture
+def distanceless_angle_constraints():
+    handler = ConstraintHandler(version=0.3)
+    handler.add_parameter(
+        {
+            "smirks": "[*:1]~[*]~[*:2]",  # all angles!
+            "id": "distanceless_angle",
+        },
+    )
+
+    return handler
+
+
+@pytest.fixture
+def angle_constraints_with_distance():
+    handler = ConstraintHandler(version=0.3)
+    handler.add_parameter(
+        {
+            "smirks": "[*:1]~[*]~[*:2]",  # all angles!
+            "id": "angle_with_distance",
+            "distance": Quantity("0.22222222 nm"),
+        },
+    )
+
+    return handler
+
+
+class TestConstraints:
+    def test_with_bonds_with_distanceless_constraints(
+        self,
+        sage,
+        distanceless_bond_constraints,
+        ethanol,
+    ):
+        """Bonds specified, constraints without distance, length taken from bonds."""
+        openmm = pytest.importorskip("openmm")
+
+        sage.deregister_parameter_handler("Constraints")
+        sage.register_parameter_handler(distanceless_bond_constraints)
+
+        # add in bond parameters which are also constrained, since it's easier and a little safer
+        # than looking up what they *should* be
+        system = sage.create_interchange(ethanol.to_topology()).to_openmm_system(add_constrained_forces=True)
+
+        assert system.getNumConstraints() == ethanol.n_bonds
+
+        constraint_distances = ValenceDict()
+        bond_distances = ValenceDict()
+
+        for constraint_index in range(system.getNumConstraints()):
+            atom1_index, atom2_index, distance = system.getConstraintParameters(constraint_index)
+
+            assert "0.123456789" not in str(distance)
+
+            constraint_distances[(atom1_index, atom2_index)] = distance
+
+        for force in system.getForces():
+            if type(force) is not openmm.HarmonicBondForce:
+                continue
```

(not blocking) This test will silently pass if there is no HarmonicBondForce.